src/treecontroller/viewconsumable.js

```diff
@@ -0,0 +1,486 @@
+	 * first already consumed item is found and `null` when first non-consumable item is found.
+	 *
+	 *		viewConsumable.test( p, { name: true } ); // Tests element's name.
+	 *		viewConsumable.test( p, { attribute: 'name' } ); // Tests attribute.
+	 *		viewConsumable.test( p, { class: 'foobar' } ); // Tests class.
+	 *		viewConsumable.test( p, { style: 'color' } ); // Tests style.
+	 *		viewConsumable.test( p, { attribute: 'name', style: 'color' } ); // Tests attribute and style.
+	 *		viewConsumable.consume( p, { class: [ 'baz', 'bar' ] } ); // Multiple consumables can be tested.
+	 *
+	 * Throws {@link utils.CKEditorError CKEditorError} `viewconsumable-invalid-attribute` when `class` or `style`
+	 * attribute is provided - it should be handled separately by providing actual style/class.
+	 *
+	 *		viewConsumable.test( p, { attribute: 'style' } ); // This call will throw an exception.
+	 *		viewConsumable.test( p, { style: 'color' } ); // This is properly handled style.
+	 *
+	 * @param {engine.treeView.Element} element
+	 * @param {Object} consumables
+	 * @param {Boolean} consumables.name If set to true element's name will be included.
+	 * @param {String|Array} consumables.attribute Attribute name or array of attribute names.
+	 * @param {String|Array} consumables.class Class name or array of class names.
+	 * @param {String|Array} consumables.style Style name or array of style names.
+	 * @returns {Boolean|null} Returns `true` when all items included in method's call can be consumed. Returns `false`
+	 * when first already consumed item is found and `null` when first non-consumable item is found.
+	 */
+	test( element, consumables ) {
+		const elementConsumables = this._consumables.get( element );
+
+		if ( elementConsumables === undefined ) {
+			return null;
+		}
+
+		return elementConsumables.test( consumables );
+	}
+
+	/**
+	 * Consumes provided {@link engine.treeView.Element view Element} and its parts.
+	 * It returns `true` when all items included in method's call can be consumed, otherwise returns `false`.
+	 *
+	 *		viewConsumable.consume( p, { name: true } ); // Consumes element's name.
+	 *		viewConsumable.consume( p, { attribute: 'name' } ); // Consumes element's attribute.
+	 *		viewConsumable.consume( p, { class: 'foobar' } ); // Consumes element's class.
+	 *		viewConsumable.consume( p, { style: 'color' } ); // Consumes element's style.
+	 *		viewConsumable.consume( p, { attribute: 'name', style: 'color' } ); // Consumes attribute and style.
+	 *		viewConsumable.consume( p, { class: [ 'baz', 'bar' ] } ); // Multiple consumables can be consumed.
+	 *
+	 * Throws {@link utils.CKEditorError CKEditorError} `viewconsumable-invalid-attribute` when `class` or `style`
+	 * attribute is provided - it should be handled separately by providing actual style/class.
+	 *
+	 *		viewConsumable.consume( p, { attribute: 'style' } ); // This call will throw an exception.
```

Comment:
It should be possible to consume or test `style` or `class` attribute. It would consume all styles or class, return false if there it any of styles/classes which is not consumable.